moveit_ros/moveit_servo/test/unit_test_servo_calcs.cpp

```diff
@@ -360,7 +360,7 @@ TEST_F(ServoCalcsTestFixture, TestEnforceVelLimits)
   for (size_t i = 0; i < 7; ++i)
   {
     // We need to check vs radians-per-loop allowable rate (not rad/s)
-    EXPECT_LE(desired_velocity[i], vel_limits[i] * servo_calcs_->parameters_->publish_period);
+    EXPECT_GE(desired_velocity[i], vel_limits[i] * servo_calcs_->parameters_->publish_period);
```

Comment:
I don't think this test condition should be flipped. After calling `enforceVelLimits()` the desired velocity should be within the limits.

Reply:
I see, I thought it was vice versa, but now makes sense. So probably this means we have something wrong in enforceVelLimits?

Reply:
Yes, seems like it. Did you compare the values? Maybe it's just a rounding error and we need to add an epsilon

Reply:
0.102 vs 0.081345 for positive
-0.102 vs -0.097614 for negative

Reply:
It would be a good idea to port this MoveIt1 PR that really simplified the limit enforcement before digging too much farther into this

https://github.com/ros-planning/moveit/pull/2471

Reply:
@AndyZe I can port it quickly if you want seem pretty straightforward to port. Do you want me to include basic functionality tests and remove (/ rename) low latency tests?

Reply:
Some of the existing tests seem just fine to me, including:

TestEnforcePosLimits
TestEnforceVelLimits

Can you link me to exactly what tests you might change?

Reply:
:+1: to porting that PR

Reply:
I meant the naming from the PR, it seems you guys have changed the naming from low_latency to basic_servo_tests

Reply:
I see. I think `basic_servo_tests` is definitely a better, more descriptive name.

Reply:
Tested in my local with the one on ros-planning/moveit#2471 though it seems to give the exact same results for `enforceVelocityLimits()`